tiatoolbox/utils/misc.py

```diff
@@ -338,6 +337,77 @@ def contrast_enhancer(img, low_p=2, high_p=98):
     return np.uint8(img_out)
 
 
+def read_point_annotations(input_table):
+    """Read annotations as pandas DataFrame.
+
+    Args:
+        input_table (str or pathlib.Path or ndarray or pd.DataFrame): path to csv, npy
+         or json or an ndarray. first column in the table represents x position, second
+         column represents y position. The third column represents the class. If the
+         table has headers, the header should be x, y & class.
+
+    Returns:
+        pd.DataFrame: DataFrame with x, y location and class type.
+
+    Examples:
+        >>> from tiatoolbox.utils.misc import read_point_annotations
+        >>> labels = read_point_annotations('./annotations.csv')
+
+    """
+    if isinstance(input_table, (str, pathlib.Path)):
+        _, _, suffix = split_path_name_ext(input_table)
+
+        if suffix == ".npy":
+            out_table = np.load(input_table)
+            if out_table.shape[1] == 2:
+                out_table = pd.DataFrame(out_table, columns=["x", "y"])
+                out_table["class"] = None
+            else:
+                out_table = pd.DataFrame(out_table, columns=["x", "y", "class"])
```

Comment:
Need strict enforcing  shape[1] == 3 here else pd will throw out error, which may be uninformative for end user, also may want to raise error or warning shape[1] > 3

Reply:
Resolved in this PR https://github.com/TIA-Lab/tiatoolbox/commit/854f4549144f618309dd275f1026120f41f41888